File: collie/src/main/java/com/snail/collie/core/ActivityStack.java
package com.snail.collie.core;

import android.app.Activity;
import android.app.Application;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.List;

public class ActivityStack {

    private static volatile ActivityStack sInstance = null;
    private List<Activity> mActivities = new ArrayList<>();
    private int mCurrentSate;

    private ActivityStack() {
    }

    public static ActivityStack getInstance() {
        if (sInstance == null) {
            synchronized (ActivityStack.class) {
                if (sInstance == null) {
                    sInstance = new ActivityStack();
                }
            }
        }
        return sInstance;
    }

    public void push(Activity activity) {
        mActivities.add(0, activity);
    }

    public void pop(Activity activity) {
        mActivities.remove(activity);
    }

    public void markResume() {
        mCurrentSate++;
    }

    public void markStop() {
        mCurrentSate--;
    }

    public Activity getTopActivity() {

        return mActivities.size() > 0 ? mActivities.get(0) : null;
    }

    public boolean isInBackGround() {
        return mCurrentSate == 0;
    }
}

File: collie/src/main/java/com/snail/collie/core/ActivityStack.kt
package com.snail.collie.core

import android.app.Activity

object ActivityStack {

    private val mActivities: MutableList<Activity> = mutableListOf()

    @Volatile
    private var mCurrentSate = 0

    fun push(activity: Activity) {
        mActivities.add(0, activity)
    }

    fun getSize(): Int {
        return mActivities.size
    }

    fun pop(activity: Activity) {
        mActivities.remove(activity)
    }

    fun markStart() {
        mCurrentSate++
    }

    fun markStop() {
        mCurrentSate--
    }

    fun getTopActivity(): Activity? {
        return   mActivities.first()
    }

    fun getBottomActivity(): Activity? {
        return mActivities.last()
    }

    fun isInBackGround(): Boolean {
        return mCurrentSate == 0
    }

}